pkg/mountpoint/args.go

```diff
@@ -0,0 +1,144 @@
+package mountpoint
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	ArgForeground      = "--foreground"
+	ArgReadOnly        = "--read-only"
+	ArgAllowOther      = "--allow-other"
+	ArgAllowRoot       = "--allow-root"
+	ArgRegion          = "--region"
+	ArgCache           = "--cache"
+	ArgUserAgentPrefix = "--user-agent-prefix"
+	ArgAWSMaxAttempts  = "--aws-max-attempts"
+)
+
+// An ArgKey represents the key of an argument.
+type ArgKey = string
+
+// An ArgValue represents the value of an argument.
+type ArgValue = string
+
+// A value to use in arguments without any value, i.e., an option.
+const ArgNoValue = ""
+
+// An Arg represents an argument to be passed to Mountpoint.
+type Arg struct {
+	key   ArgKey
+	value ArgValue
+}
+
+// String returns string representation of the argument to pass Mountpoint.
+func (a *Arg) String() string {
+	if a.value == ArgNoValue {
+		return a.key
+	}
+	return fmt.Sprintf("%s=%s", a.key, a.value)
+}
+
+// An Args represents arguments to be passed to Mountpoint during mount.
+type Args struct {
+	args sets.Set[Arg]
+}
+
+// ParseArgs parses given list of unnormalized and returns a normalized [Args].
+func ParseArgs(passedArgs []string) Args {
+	args := sets.New[Arg]()
+
+	for _, arg := range passedArgs {
+		var key, value string
+
+		parts := strings.SplitN(strings.Trim(arg, " "), "=", 2)
+		if len(parts) == 2 {
+			// Ex: --key=value, key=value
+			key, value = parts[0], parts[1]
+		} else {
+			// Ex: --key value, key value
```

Comment:
This section doesn't make sense to me what it's meant to catch. Don't we only support one key/value pair per argument?

Reply:
Yes we only support one argument per line, the comments lists some example usages which would fall into the associated blocks. Here, this block is if the user provide arguments like: `--key value` or `key value` (without `--` prefix) without using `=` between key and value.

Reply:
Ah, could you make the comments '`--key value`, or `key value`'? Or some other way to make it obvious the comma isn't part of the example

Reply:
Addressed with https://github.com/awslabs/mountpoint-s3-csi-driver/pull/349/commits/21a0fe0d891ae509f46d60aef81a05c514e179a5